### DISCUSSION TITLE
Terraform Module Best Practice: 1 Repo & X Sub-Folders or X repos & 0 Sub-Folders?

### DISCUSSION THREAD

**Discussion by skkc2** [gruntwork-io/knowledge-base#479] [Help] [answered]
Hi Team,

Is it better to have one Terraform repo with lots of sub-folders (i.e. modules like in **https://github.com/gruntwork-io/terragrunt-infrastructure-modules-example**) OR lots of seperate terraform module repos with no sub-folders?

Thanks in advance

---

<ins datetime="2022-06-22T10:10:51Z">
  <p><a href="https://support.gruntwork.io/hc/requests/108811">Tracked in ticket #108811</a></p>
</ins>

**Comment by yorinasub17** [comment, marked as answer]
You can refer to the tradeoffs mentioned in [the root README of that repo](https://github.com/gruntwork-io/terragrunt-infrastructure-modules-example#monorepo-vs-polyrepo) for thoughts and considerations on this topic.
